### ISSUE TITLE
[Chore]: Update aws sdk to the latest version

### ISSUE THREAD

**Issue by kibertoad** [bbc/sqs-consumer#509]
### Describe the bug

AWS SDK has introduced memory leak while updating one of the dependencies, namely due to this change: https://github.com/smithy-lang/smithy-typescript/commit/c2a5595078e169ea358c379bd4b895d13a05dea2

See the bug issue: https://github.com/smithy-lang/smithy-typescript/pull/1332

The problematic version of the dependency was introduced in AWS SDK in 3.609.0, due to smithy dependency updates, see https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.609.0

For that particular reason, latest peerDependency of `sqs-consumer` is rather unfortunate:
```
"peerDependencies": {
    "@aws-sdk/client-sqs": "^3.609.0"
  },
```

There is a significant risk that users, unaware of the memory leak, may use one of the affected versions, due to the current peerDependency range.

This issue was resolved in the latest version, see https://github.com/aws/aws-sdk-js-v3/compare/v3.613.0...v3.614.0#diff-1fc4f6a20ec76cddaa471b6988d57674be4ae65d3d2dbe6ac5e1645bd4ba8fadR47

I would strongly recommend to bump peerDependency version to 3.614.0 within a _patch_ release, in order to eliminate the memory leak risk.

### Your minimal, reproducible example

-

### Steps to reproduce

Any SQS consumer running on a version between 3.609 to 3.613 will be consistently leaking memory over time.

### Expected behavior

No memory leaks occur as SQS consumer is polling for messages.

### How often does this bug happen?

Every time

### Platform

Any OS

### Package version

10.3.0

### AWS SDK version

3.609

**Comment by nicholasgriffintn** [comment]
Because we use a ^, it should download this version for anyone installing since the update, that said, people should manually check which we noted in the release.

AWS also uses a ^ for this package, so this version would also download the latest version.

So it shouldn't be that wide of an issue.

That said, we can certainly release a patch with the new version.

**Comment by kibertoad** [comment]
@nicholasgriffintn That's true, which is why it's a risk and not a guaranteed memory leak. However, if someone updated to affected version specifically and stored pinned version in a lockfile, they would have a problem.

I'm not saying that sqs-consumer is making the problem worse, just that it can help to alleviate it at a low cost.

And affected versions _definitely_ shouldn't be used in prod. We noticed the problem because our services started to OOM at a regular basis.

**Comment by nicholasgriffintn** [comment]
Agreed. Just released 11.0.1 with the sdk version update. Links above ^.

**Comment by kibertoad** [comment]
thank you!
